subsys/net/l2/ethernet/dsa/dsa_port.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_dsa_port, CONFIG_NET_DSA_LOG_LEVEL);
+
+#include <zephyr/net/dsa_core.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/net/phy.h>
+
+#ifdef CONFIG_NET_L2_ETHERNET
+bool dsa_port_is_conduit(struct net_if *iface)
+{
+	/* First check if iface points to ETH interface */
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(ETHERNET)) {
+		/* Check its capabilities */
+		if (net_eth_get_hw_capabilities(iface) & ETHERNET_DSA_CONDUIT_PORT) {
+			return true;
+		}
+	}
+
+	return false;
+}
+#else
+bool dsa_port_is_conduit(struct net_if *iface)
+{
+	return false;
+}
+#endif
+
+int dsa_port_initialize(const struct device *dev)
+{
+	const struct dsa_port_config *cfg = dev->config;
+	struct dsa_context *dsa_ctx = dev->data;
+	int err = 0;
+
+	dsa_ctx->init_ports++;
+
+	if (cfg->pincfg) {
+		err = pinctrl_apply_state(cfg->pincfg, PINCTRL_STATE_DEFAULT);
+		if (err) {
+			goto out;
+		}
+	}
+
+	if (dsa_ctx->dapi->port_init) {
+		err = dsa_ctx->dapi->port_init(dev);
+		if (err) {
+			goto out;
+		}
+	}
+
+out:
+	/* All ports are initialized. May need switch setup. */
+	if (dsa_ctx->init_ports == dsa_ctx->num_ports) {
+		if (dsa_ctx->dapi->switch_setup) {
+			err = dsa_ctx->dapi->switch_setup(dsa_ctx);
+		}
+	}
+
+	return err;
+}
+
+static void dsa_port_iface_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct dsa_port_config *cfg = (struct dsa_port_config *)dev->config;
+	struct dsa_context *dsa_ctx = dev->data;
+	struct ethernet_context *eth_ctx = net_if_l2_data(iface);
+	struct ethernet_context *eth_ctx_conduit = NULL;
+	int i = cfg->port_idx;
+
+	/* Find conduit port */
+	if (!dsa_ctx->iface_conduit && cfg->ethernet_connection) {
+		dsa_ctx->iface_conduit = net_if_lookup_by_dev(cfg->ethernet_connection);
+		if (dsa_ctx->iface_conduit == NULL) {
+			LOG_ERR("DSA: Conduit iface NOT found!");
+			return;
+		}
+
+		/*
+		 * Provide pointer to DSA context to conduit's eth interface
+		 * struct ethernet_context
+		 */
+		eth_ctx_conduit = net_if_l2_data(dsa_ctx->iface_conduit);
+		eth_ctx_conduit->dsa_ctx = dsa_ctx;
+		eth_ctx_conduit->dsa_port = DSA_CONDUIT_PORT;
+	}
+
+	if (strcmp(cfg->phy_mode, "internal") == 0) {
```

Comment:
Sometimes external security checker tools might start to complain about these "unsafe" string functions. So in order to avoid that in the future, we could be proactive here and change this to
```
if (strncmp(cfg->phy_mode, "internal", sizeof("internal") - 1) == 0) {
```

Reply:
Thanks, I will convert to use that.